Fix dietpi-logclear failing on files in sticky bit directories

Timoses opened the pull request:
fixes #6506

Timoses committed:
Fix dietpi-logclear failing on files in sticky bit directories

fixes #6506

Timoses committed:
second approach

Timoses committed:
fix wording

Timoses commented on dietpi/func/dietpi-logclear:
```diff
@@ -142,6 +143,9 @@
 		done
 
+		# Truncate files that failed with bash '>' redirect
+		truncate -c -s0 ${ARRAY_LOGFILE_TRUNCATE_DENIED[*]}
```
Should probably add a `[ ${#key[@]} -gt 0 ] && truncate ...` ?

MichaIng replied:
Makes sense. Also calling the array as array and quoting the elements, in case there are special characters or spaces within the path/file names:
```suggestion
		(( ${#ARRAY_LOGFILE_TRUNCATE_DENIED[@]} )) && truncate -c -s0 "${ARRAY_LOGFILE_TRUNCATE_DENIED[@]}"
```

MichaIng commented on dietpi/func/dietpi-logclear:
```diff
@@ -117,13 +118,13 @@
 
 					# Clear log file contents
-					> "$i"
+					2> /dev/null > "$i" || ARRAY_LOGFILE_TRUNCATE_DENIED+=("$i")
```
I think the STDERR redirect like that does nothing? Did you want to mute possible errors messages of the STDOUT redirect? I think that would only work like this:
```sh
{ > "$i"; } 2> /dev/null
```
But as I mentioned in the issue, I am no fans of redirection STDERR, as long as we do not always expect errors. There might be important errors, like hardware I/O issues, filesystem corruption and whatnot, what one should see. So I prefer to completely switch to `truncate`:
```suggestion
					ARRAY_LOGFILE_TRUNCATE_DENIED+=("$i")
```
Could be named differently then, of course.

Timoses replied:
`2> /dev/null > "$i"` actually does work in muting stderr (tested). `> "$i" 2> /dev/null` does not work.

MichaIng commented on dietpi/func/dietpi-logclear:
```diff
@@ -117,13 +118,13 @@
 				elif (( $INPUT == 1 )); then
 
-					> "$i"
+					2> /dev/null > "$i" || ARRAY_LOGFILE_TRUNCATE_DENIED+=("$i")
```
```suggestion
					ARRAY_LOGFILE_TRUNCATE_DENIED+=("$i")
```

MichaIng committed:
v8.20

- DietPi-Logclear | Truncate all log files which are to be cleared, and double quote individual file paths collected in the array

MichaIng committed:
v8.20

- CHANGELOG | DietPi-Logclear: Resolved an issue where clearing files within sticky bit directories failed. Many thanks to @Timoses for fixing this issue: https://github.com/MichaIng/DietPi/pull/6507
